File: Graph-imp/cpp/DFS.cpp
#include<iostream>
#include<vector>
using namespace std;

class DFS{

    void dfs(int node, vector<int> &visited, vector<int> adj[], vector<int> &storeDfs){
          storeDfs.push_back(node);

          for(auto it: adj[node]){
              if(!visited[it]){
                  visited[it] = 1;
                  dfs(it, visited, adj, storeDfs);
              }
          }

    }

  public:
     vector<int> dfsOfGraph(int node, vector<int> adj[]){
              
              vector<int> storeDfs;
              vector<int> visited(node+1,0);

              for(int i=1; i<=node; i++){
                  if(!visited[i]){
                      visited[i] = true;
                      dfs(i, visited,adj, storeDfs);
                  }
              }

              return storeDfs;
      }
};


int main(){
    int n,e;
    cin>>n>>e;

    vector<int> adj[n+1];
    
    for(int i=0; i<e; i++){
        int u,v;
        cin>>u>>v;

        adj[u].push_back(v);
        adj[v].push_back(u);
    }

     DFS objDFS;

     vector<int> resutlOfDfs = objDFS.dfsOfGraph(n,adj);
     
     for(auto it: resutlOfDfs){
         cout<<it<<" ";
     }

}

File: Graph-imp/java/DFS.java
import java.util.*;


class DFS {

    public static void main(String[] args) {
       Scanner sc = new Scanner(System.in);
       int n = sc.nextInt();
       int e = sc.nextInt();

       ArrayList<ArrayList<Integer>> adj = new ArrayList<ArrayList<Integer>>(n+1);
       for(int i=0; i<n+1; i++)
         adj.add(new ArrayList<Integer>());

       for(int i=0; i<e; i++){
           int  u = sc.nextInt();
           int  v = sc.nextInt();

           adj.get(u).add(v);
           adj.get(v).add(v);
       }

       Sol obj = new Sol();

       ArrayList<Integer> result = new ArrayList<>();

       result = obj.dfsOfGraph(n, adj);

       for(Integer it: result){
           System.out.print(it+" ");
       }
    }
       
}



class Sol{

    void dfs(int node, boolean visited[], ArrayList<ArrayList<Integer>> adj, ArrayList<Integer> storeDfs ){
             storeDfs.add(node);
             for(Integer it: adj.get(node)){
                if(visited[it] == false){
                    visited[it] = true;
                    dfs(it, visited, adj, storeDfs);
                }
             }
    }

    public ArrayList<Integer> dfsOfGraph(int node, ArrayList<ArrayList<Integer>> adj){
        ArrayList<Integer> storeDfs = new ArrayList<>();
        boolean[] visited = new boolean[node+1];
    
        for(int i=1; i<=node; i++){
            if(visited[i] == false){
                 visited[i] = true;
                 dfs(i,visited, adj, storeDfs);
            }
        }

      return storeDfs;
    }
}
